cli/src/cli_util.rs

```diff
@@ -481,6 +481,12 @@ impl From<WorkingCopyStateError> for CommandError {
     }
 }
 
+impl From<GitIgnoreError> for CommandError {
+    fn from(err: GitIgnoreError) -> Self {
+        user_error_with_message("Failed to process .gitignore.", err)
```

Comment:
What do we do when this happens? What does Git do in the same cases? It seem like we shouldn't fail to snapshot the working copy. Maybe we should report a bug to the `insta` crate if that's what happens in some cases where Git doesn't report an error.

Reply:
Actually it only come from non ignore crate code during parse path.
(chain and chain_with_file  function in `lib/src/gitignore.rs`
``` rust
pub fn chain(
        self: &Arc<GitIgnoreFile>,
        prefix: &str,
        input: &[u8],
    ) -> Result<Arc<GitIgnoreFile>, GitIgnoreError> {
        let path = self.path.clone() + prefix;
        let mut builder = gitignore::GitignoreBuilder::new(&path);
        for (i, input_line) in input.split(|b| *b == b'\n').enumerate() {
            let line =
                std::str::from_utf8(input_line).map_err(|err| GitIgnoreError::InvalidUtf8 {
                    path: PathBuf::from(&path),
                    line_num_for_display: i + 1,
                    line: String::from_utf8_lossy(input_line).to_string(),
                    source: err,
                })?;
```

```rust
pub fn chain_with_file(
        self: &Arc<GitIgnoreFile>,
        prefix: &str,
        file: PathBuf,
    ) -> Result<Arc<GitIgnoreFile>, GitIgnoreError> {
        if file.is_file() {
            let buf = fs::read(&file).map_err(|err| GitIgnoreError::ReadFile {
                path: file.clone(),
                source: err,
            })?;
```

Reply:
I see. Then I think we should change our code to avoid returning an error in this case. We would ideally match Git's behavior. Seems fine to attempt to read the entire file as utf8 and ignore it if that fails. Or check exactly what git does if you feel like it, but I'm not going to insist.

Reply:
Got it. I guess git just treat as a binary string.

```console
# create utf8 file.
daehyeok@macbook:~/temp$ echo "파일"  > "파일"
# Git status showing hex codes instead of utf8 string.
daehyeok@macbook:~/temp$ git status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	"\355\214\214\354\235\274"

nothing added to commit but untracked files present (use "git add" to track)
daehyeok@macbook:~/temp$
# Write utf8 path to gitginore
daehyeok@macbook:~/temp$ echo "파일"  > .gitignore
# Check it's work or not.
daehyeok@macbook:~/temp$ git status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.gitignore

nothing added to commit but untracked files present (use "git add" to track)
daehyeok@macbook:~/temp$
# Let's see what happen when gitignore contain non utf8(euckr)
daehyeok@macbook:~/temp$ iconv -f utf8 -t euc_kr 파일  > ./.gitignore
daehyeok@macbook:~/temp$ iconv -f utf8 -t euckr 파일 | xargs git check-ignore
"\306\304\300\317"
```

Reply:
However, jj only allows UTF-8 paths, so we already can't do "what Git does"...

Unlike Git, if we fail to ignore something (if, upon encountering an incomprehensible pattern in a `.gitignore` file, we simply logged a warning and proceeded with snapshotting anyways), we run the risk of accidentally committing secrets or excessively-large files. For Git, it would be fine to log a warning and proceed, because the user still has to actively stage and commit changes to those files.

Reply:
>  we run the risk of accidentally committing secrets or excessively-large files. 

`chain` and `chain_with_file` only used to parse .gitignore file. so it will not change current JJ behavior. is it right?
 Warning/ignore then process might need to use JJ with legacy git repo.
 
 Also current version(0.14) already progressed with non utf8 gitignore.

Reply:
hmm How can I write warning log and progressed on `lib` codes?

Reply:
Good point that we should be careful to not commit files that were supposed to be ignored. So erroring out as you had done sounds good then, @daehyeok. I think that means this PR is ready to merge, right?

Reply:
> Also current version(0.14) already progressed with non utf8 gitignore.

Yes, that's correct, but maybe we shouldn't have done that, as @arxanas noted.

Reply:
> Good point that we should be careful to not commit files that were supposed to be ignored. So erroring out as you had done sounds good then, @daehyeok. I think that means this PR is ready to merge, right?

yes, it we don't have any other concern  :blush: